### PR TITLE
[libwebsockets] version 4.3.3

### PR DESCRIPTION
**Pull request by maxfadson**
Hi! I’ve added a build script for [libwebsockets](https://github.com/warmcat/libwebsockets) version 4.3.3, I think it will be useful :)

### CODE REVIEW

**Comment by maxfadson** [comment]
Gonna try updating OpenSSL to 3.0.8, then I’ll reopen the PR

File: L/libwebsockets/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency("OpenSSL_jll"; compat="3.0.8"),
+    Dependency("Zlib_jll")
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6", preferred_gcc_version = v"5.2.0")
```
Comment (giordano) [inline L/libwebsockets/build_tarballs.jl:47]: Missing newline:
```suggestion
build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6", preferred_gcc_version = v"5.2.0")

```

File: L/libwebsockets/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    -DLWS_WITH_HTTP2=1 \
+    -DLWS_WITHOUT_TESTAPPS=1
+
+make -j${nproc}
+make install
+""" 
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms(exclude = x -> (Sys.iswindows(x)))
```
Comment (giordano) [inline L/libwebsockets/build_tarballs.jl:33]:
```suggestion
platforms = supported_platforms(exclude = Sys.iswindows)
```

File: L/libwebsockets/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+cmake .. -DCMAKE_INSTALL_PREFIX=${prefix} \
+    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DZLIB_LIBRARY=${libdir}/libz.${dlext} \
+    -DZLIB_INCLUDE_DIR=${includedir} \
+    -DLWS_WITH_HTTP2=1 \
+    -DLWS_WITHOUT_TESTAPPS=1
+
+make -j${nproc}
+make install
```
Comment (giordano) [inline L/libwebsockets/build_tarballs.jl:28]: https://docs.binarybuilder.org/stable/build_tips/#CMake-builds
```suggestion
cmake -B build \
    -DCMAKE_INSTALL_PREFIX=${prefix} \
    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
    -DCMAKE_BUILD_TYPE=Release \
    -DZLIB_LIBRARY=${libdir}/libz.${dlext} \
    -DZLIB_INCLUDE_DIR=${includedir} \
    -DLWS_WITH_HTTP2=1 \
    -DLWS_WITHOUT_TESTAPPS=1

cmake --build build --parallel ${nproc}
cmake --install build
```

File: L/libwebsockets/build_tarballs.jl [hunk 4]
```diff
@@ -0,0 +1,47 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message
+using BinaryBuilder, Pkg
+
+name = "libwebsockets"
+version = v"4.3.3"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/warmcat/libwebsockets.git", "5102a5c8d6110b25a01492fcf96fb668b13dd6e7"),
```
Comment (giordano) [inline L/libwebsockets/build_tarballs.jl:10]: That commit isn't the commit corresponding to v4.3.3: https://github.com/warmcat/libwebsockets/commit/4415e84c095857629863804e941b9e1c2e9347ef

Comment (maxfadson) [reply L/libwebsockets/build_tarballs.jl]: Oh, I accidentally grabbed the latest hash from the 4.3-stable branch 😅 
Fixed it now by adding the 4.3.3 tag hash!

**Comment by giordano** [comment]
Thanks for your contribution!

**Pull request merged**